### PR TITLE
Ported the TCP proxy into the X-Ray receiver

### LINKED ISSUE #750: Enable centralized sampling in X-Ray receiver
Port the existing TCP proxy server from the X-Ray daemon into the X-Ray receiver in order to support centralized sampling

### PR DESCRIPTION
**Pull request by hencrice**
**Description:** 
Ported the TCP proxy from the X-Ray daemon into the X-Ray receiver in order to support the centralized sampling feature.

This PR is identical to #763. Only created a new one and rebased on top of master because codecov was busted on the original PR.

**Link to tracking Issue:**
Closes #750

**Testing:**
Existing unit tests updated. New tests added

**Documentation:**
N/A

### CODE REVIEW

**Comment by hencrice** [comment]
codecov complained about "Missing base report. Unable to compare commits because the base of the pull request did not upload a coverage report."

**Comment by hencrice** [comment]
Based on the error generated by codecov https://codecov.io/gh/open-telemetry/opentelemetry-collector-contrib/pull/774

I will rebase this branch on top of current master

**Comment by hencrice** [comment]
Somehow `lint` seems very unstable on CircleCI

**Comment by bogdandrutu** [comment]
@anuraaga please review, you are AWS exporter expert :)

File: receiver/awsxrayreceiver/go.mod [hunk 1]
```diff
@@ -5,13 +5,9 @@ go 1.14
 replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/common => ../../internal/common
 
 require (
-	github.com/aws/aws-sdk-go v1.34.5
+	github.com/aws/aws-sdk-go v1.34.8
 	github.com/google/uuid v1.1.1
-	github.com/googleapis/gnostic v0.4.1 // indirect
 	github.com/stretchr/testify v1.6.1
-	go.opentelemetry.io/collector v0.8.1-0.20200819173546-64befbcc0564
+	go.opentelemetry.io/collector v0.8.0
```
Comment (anuraaga) [inline receiver/awsxrayreceiver/go.mod:11]: @bogdandrutu Is this version right?

**Review by anuraaga — APPROVED** [review]
Code LGTM, thanks! One question about the collector dependency, I believe we're not supposed to downgrade it but not sure

File: receiver/awsxrayreceiver/go.mod (hunk 1 above)
Comment (bogdandrutu) [reply receiver/awsxrayreceiver/go.mod:11]: No we should use the one that is there already and in all the targets

**Comment by hencrice** [comment]
@bogdandrutu Reverted the accidental version downgrade. Can be merged now.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**